cmd/talosctl/cmd/mgmt/cluster/create/clustermaker/maker_test.go

```diff
@@ -0,0 +1,461 @@
+		input := getTestOps()
+		input.SkipK8sNodeReadinessCheck = true
+		input.SkipKubeconfig = true
+
+		getFinalizedClusterMaker(t, input)
+	})
+
+	addFieldTest("", n(&tf, &tf.NetworkIPv6), func(t *testing.T) {
+		input := getTestOps()
+		input.NetworkIPv6 = true
+
+		cm := getFinalizedClusterMaker(t, input)
+
+		nodes := slices.Concat(cm.request.Nodes.ControlPlaneNodes(), cm.request.Nodes.WorkerNodes())
+
+		for _, n := range nodes {
+			assert.Equal(t, 2, len(n.IPs))
+			assert.Equal(t, true, n.IPs[1].Is6())
+		}
+
+		assert.Equal(t, "10.5.0.2", nodes[0].IPs[0].String())
+		assert.Equal(t, "fd74:616c:a05::2", nodes[0].IPs[1].String())
+		assert.Equal(t, "10.5.0.5", nodes[3].IPs[0].String())
+		assert.Equal(t, "fd74:616c:a05::5", nodes[3].IPs[1].String())
+	})
+	addFieldTest("", n(&tf, &tf.InputDir), func(t *testing.T) {
+		dir := t.TempDir()
+		input := getTestOps()
+		input.RootOps.StateDir = dir
+		input.RegistryMirrors = []string{"test.test=https://test.mirror"}
+		cm := getFinalizedClusterMaker(t, input)
+
+		err := cm.configBundle.Write(dir, encoder.CommentsDisabled, machine.TypeControlPlane, machine.TypeWorker)
+		assert.NoError(t, err)
+
+		input = getTestOps()
+		input.InputDir = dir
+		cm = getFinalizedClusterMaker(t, input)
+
+		assert.EqualValues(t, "https://test.mirror", cm.request.Nodes.WorkerNodes()[0].Config.RawV1Alpha1().MachineConfig.Registries().Mirrors()["test.test"].Endpoints()[0])
+		assert.EqualValues(t, "https://test.mirror", cm.request.Nodes.ControlPlaneNodes()[0].Config.RawV1Alpha1().MachineConfig.Registries().Mirrors()["test.test"].Endpoints()[0])
+	})
+}
+
+func getFinalizedClusterMaker(t *testing.T, input Options) clusterMaker {
+	cm, err := newClusterMaker(Input{input, testProvisioner{}, testTalosVersion})
+	assert.NoError(t, err)
+	err = cm.finalizeRequest()
+	assert.NoError(t, err)
+
+	return cm
+}
+
+func getGenOpts(t *testing.T, input Options) generate.Options {
+	cm, err := newClusterMaker(Input{input, testProvisioner{}, testTalosVersion})
+	assert.NoError(t, err)
+	err = cm.finalizeRequest()
+	assert.NoError(t, err)
+	result, err := generate.NewInput(input.RootOps.ClusterName, "cluster.endpoint", "k8sv1", cm.genOpts...)
+	assert.NoError(t, err)
+
+	return result.Options
+}
+
+func addFieldTest(testName string, fieldNames []string, test func(t *testing.T)) {
+	testedFields = append(testedFields, fieldNames...)
+	name := fmt.Sprintf("Test%sField", fieldNames[0])
+
+	if testName != "" {
+		name = testName
+	}
+
+	if len(fieldNames) > 1 && testName == "" {
+		panic("no test name specified with multiple fields tested")
+	}
+
+	fieldTests = append(fieldTests, fieldTest{fieldNames, test, name})
+}
+
+var fieldTests []fieldTest
+
+type fieldTest = struct {
+	fieldNames []string
+	test       func(t *testing.T)
+	name       string
+}
+
+var testedFields = []string{}
+
+func TestAllOptionFields(t *testing.T) {
```

Comment:
Test that there are tests for all the common fields.

Please let me know if this is inappropriate or if there's a less hacky way to achieve the same result or if I just change it to good ol' fashioned tests